lib/msal-node/test/client/ManagedIdentitySources/AzureArc.spec.ts

```diff
@@ -110,6 +119,62 @@ describe("Acquires a token successfully via an Azure Arc Managed Identity", () =
             );
         });
 
+        test("acquires a token when both/either the identityEndpoint and/or imdsEndpoint environment variables are undefined, and the himds executable exists and its permissions allow it to be read", async () => {
+            // delete the environment variables so the himds executable is checked
+            delete process.env[
+                ManagedIdentityEnvironmentVariableNames.IDENTITY_ENDPOINT
+            ];
+            delete process.env[
+                ManagedIdentityEnvironmentVariableNames.IMDS_ENDPOINT
+            ];
+            // delete value cached from getManagedIdentitySource() in the beforeEach
+            delete ManagedIdentityClient["sourceName"];
+
+            // MI source will not be Azure Arc yet, since the environment variables are undefined,
+            // and accessSyncSpy still returns an error
+            // (meaning either the himds file doesn't exists or its permissions don't allow it to be read)
+            expect(
+                managedIdentityApplication.getManagedIdentitySource()
+            ).not.toBe(ManagedIdentitySourceNames.AZURE_ARC);
+            // delete value cached from getManagedIdentitySource() directly above
+            delete ManagedIdentityClient["sourceName"];
+
+            // returns undefined when the himds file exists and its permissions allow it to be read
+            // otherwise, throws an error
+            accessSyncSpy.mockImplementationOnce(() => {
+                return undefined;
+            });
+
+            expect(managedIdentityApplication.getManagedIdentitySource()).toBe(
+                ManagedIdentitySourceNames.AZURE_ARC
+            );
+
+            // returns undefined when the himds file exists and its permissions allow it to be read
+            // otherwise, throws an error
+            accessSyncSpy.mockImplementationOnce(() => {
+                return undefined;
+            });
+            const networkManagedIdentityResult: AuthenticationResult =
+                await managedIdentityApplication.acquireToken(
+                    managedIdentityRequestParams
+                );
+            expect(networkManagedIdentityResult.fromCache).toBe(false);
+
+            expect(networkManagedIdentityResult.accessToken).toEqual(
+                DEFAULT_SYSTEM_ASSIGNED_MANAGED_IDENTITY_AUTHENTICATION_RESULT.accessToken
+            );
+
+            // one for each call to getManagedIdentitySource() + one for the acquireToken call
+            expect(accessSyncSpy).toHaveBeenCalledTimes(3);
+
+            // reset the environment variables to expected values for Azure Arc tests
+            process.env[
```

Comment:
Why are you resetting these?

Reply:
All other azure arc tests need those environment variables; I set them to these values at the top of this file. They need to deleted for this test so that the trigger for checking for himds exe happens, then they need to be re-set to the expected values.